lib/Idream/Effects/Idris.hs

```diff
@@ -89,18 +96,24 @@ runIdris f = interpretM g where
         idrisArgs = [ "--libdir" ]
         toDir = filter (/= '\n')
     toDir <$> invokeIdrisWithEnv eh1 eh2 idrisArgs Nothing []
-  g (IdrisCompile projName pkgName) = do
+  g (IdrisCompileLike projName pkgName action) = do
```

Comment:
Maybe it's better if we create a helper function which the compile and repl case then can call? I was already thinking about refactoring this big function.